admin/acceptance.py

```diff
@@ -195,6 +203,23 @@ def ensure_keys(reactor):
             found.  Otherwise, fails with ``AgentNotFound`` or ``KeyNotFound``.
         """
 
+    def extend_cluster(reactor, cluster, count, tag, starting_index):
+        """
+        Create new nodes and add them to the given cluster.
+
+        :param reactor: Reactor to use.
+        :param Cluster cluster: The cluster.
+        :param int count: How many new nodes to create.
+        :param str tag: Tag used for naming the new nodes.
+        :param int starting_index: A number used for naming the new nodes.
+        :return: List of ``Deferred``s that fire when a new node is configured
+            or fails.
+
+        .. note::
+            The cluster object is modified by this call and, thus, it should be
+            mutable
+        """
```

Comment:
We use `pyrsistent` to have immutable objects. In particular, `Cluster` is a `PClass` and its node lists are typically `pvector`s. In this case we require that they are regular lists.